packages/client/src/createTimeseriesProperty.ts

```diff
@@ -28,117 +28,70 @@ import {
   parseStreamedResponse,
 } from "./util/streamutils.js";
 
-export function createTimeseriesProperty<T extends number | string>(
-  client: MinimalClient,
-  objectApiName: string,
-  primaryKey: any,
-  propertyName: string,
-): TimeSeriesProperty<T> {
-  const getFirstPoint = async () => {
+export class TimeSeriesPropertyImpl<T extends number | string>
```

Comment:
Hmm ok, so you're making this a class instead. Ok yea still confused what the benefit here is